02-microarray/pathway-analysis_microarray_03_gsea.Rmd

```diff
@@ -0,0 +1,493 @@
+Genes are ranked from most highly positive to most highly negative and weighted according to their gene-level statistic.
+A running score is calculated by starting with the most highly ranked genes and increasing the score when a gene is in the pathway and decreasing the score when a gene is not in the pathway.
+The ES is the maximum deviation from zero.
+Significance is assessed by generating a null distribution by sampling random gene sets of the same size and an FDR (false discovery rate) value is calculated to account for multiple hypothesis testing.
+Normalized enrichment scores (NES) are enrichment scores that are scaled to make gene sets that contain different number of genes comparable [@Subramanian2005; @Korotkevich2019].
+
+We can use the `GSEA()` function to perform GSEA with any generic set of gene sets, but there are several functions for using specific, commonly used gene sets (e.g., `gseKEGG()`).
+
+```{r}
+gsea_results <- GSEA(
+  geneList = t_vector, # ordered ranked gene list
+  minGSSize = 25, # minimum gene set size
+  maxGSSize = 500, # maximum gene set set
+  pvalueCutoff = 0.05, # p value cutoff
+  eps = 0, # boundary for calculating the p value
+  seed = TRUE, # set seed to make results reproducible
+  pAdjustMethod = "BH", # Benjamini-Hochberg correction
+  TERM2GENE = dplyr::select(
+    dr_hallmark_df,
+    gs_name,
+    entrez_gene
+  )
+)
+```
+
+Let's take a look at the results.
+
+```{r}
+# We can access the results from our gseaResult object using `@result`
+head(gsea_results@result)
+```
+
+Looks like we have gene sets returned as significant at FDR of `0.05`.
+If we didn't have any, our visualizations below would show up blank as nothing would have met our `pvalueCutoff` above.
+
+The information we're most likely interested in is in the `results` slot.
+Let's convert this into a data frame that we can use for further analysis and write to file.
+
+```{r}
+gsea_result_df <- data.frame(gsea_results@result)
+```
+
+## Visualizing results
+
+We can visualize GSEA results for individual pathways or gene sets using `enrichplot::gseaplot()`.
+Let's take a look at 2 different pathways -- one with a highly positive NES and one with a highly negative NES -- to get more insight into how ES are calculated.
+
+### Highly Positive NES
+
+Let's look for the gene set with the highest positive NES.
+
+```{r}
+gsea_result_df %>%
+  # Combining the `dplyr::desc()` function with `dplyr::arrange()` sorts the
+  # values in descending order
+  dplyr::arrange(dplyr::desc(NES))
+```
+
+The gene set `HALLMARK_TNFA_SIGNALING_VIA_NFKB` has the highest NES score.
+
+```{r}
+enrichplot::gseaplot(gsea_results,
+  geneSetID = "HALLMARK_TNFA_SIGNALING_VIA_NFKB",
+  title = "HALLMARK_TNFA_SIGNALING_VIA_NFKB",
+  color.line = "#0d76ff"
+)
+```
+
+Notice how the genes that are in the gene set, indicated by the black bars, tend to be on the left side of the graph indicating that they have positive gene-level scores.
```

Comment:
I like this sentence, can we add a sentence like it for describing the bottom panel? Aka what's the red dashed line about (looks like its the peak or max deviation from 0) we should probably highlight that and what it means.